app/models/permission.rb

```diff
@@ -23,4 +23,21 @@ def self.with_translations
   def self.reset_resources
     @all_resources = nil
   end
+
+  # converts klass to name used as resource_type in permissions table
+  def self.resource_name(klass)
+    return 'Operatingsystem' if klass <= Operatingsystem
+    return 'ComputeResource' if klass <= ComputeResource
+    return 'Parameter' if klass <= Parameter
```

Comment:
note for the reviewer: I already applied https://github.com/theforeman/foreman/pull/3984/files#diff-2a8c1ced7ef4dfe08f35154c5a3ef0f1R169 since this is moving to different file and wouldn't cause conflict